src/lightning_app/core/flow.py

```diff
@@ -142,6 +142,14 @@ def __setattr__(self, name: str, value: Any) -> None:
                 if name in self._works and value != getattr(self, name):
                     raise AttributeError(f"Cannot set attributes as the work can't be changed once defined: {name}")
 
+            if isinstance(value, (list, dict)) and value:
```

Comment:
```suggestion
            if isinstance(value, (list, dict)):
```
is this needed?

Reply:
Yes, it needs to be. As empty list would trigger.